auslib/db.py

```diff
@@ -1539,18 +1539,28 @@ def _buildIDMatchesRule(self, ruleBuildID, queryBuildID):
             return True
         return string_compare(queryBuildID, ruleBuildID)
 
-    def _csvMatchesRule(self, ruleString, queryString, substring=True):
-        """Decides whether a column from a rule matches an incoming one.
-           Some columns in a rule may specify multiple values delimited by a
-           comma. Once split we do a full or substring match against the query
-           string. Because we support substring matches, there's no need
-           to support globbing as well."""
+    def _simpleBooleanMatchesSubRule(self, subRuleString, queryStringList):
+        for rule in subRuleString:
+            if rule not in queryStringList:
+                return False
+        return True
+
+    def _simpleBooleanMatchesRule(self, ruleString, queryString):
+        """Decides whether a column from a rule matches an incoming one using simplified boolean logic.
+        Only two operators are supported: '&&' (and), ',' (or). A rule like 'AMD,SSE' will match incoming
+        rules that contain either 'AMD' or 'SSE'. A rule like 'AMD&&SSE' will only match incoming rules
+        that contain both 'AMD' and 'SSE'.
+        Rules are matched to full strings, so if we have a rule that specifies 'AMD,SSE', it will NOT be
+        matched by an incoming rule that contains 'SSE3'.
+        """
         if ruleString is None:
             return True
-        for part in ruleString.split(','):
-            if substring and part in queryString:
-                return True
-            elif part == queryString:
+
+        decomposedRules = [[rule.strip() for rule in subRule.split('&&')] for subRule in ruleString.split(',')]
+        decomposedQueryString = queryString.split(',')
```

Comment:
Currently, splitting the queryString allows us to toggle between substring and full string matching using the 'in' operator. I changed this part to accommodate both types of matching. When we want to do substring matching, we leave the queryString as a string so that the 'in' operator matches any substring in the queryString. But if we want exact matching, we split the string into a list, so that the 'in' operator only returns True if the exact string appears in the resulting list. 

You're right about this being future-proof if we decide to match against multiple parts of SYSTEM_REQUIREMENTS, even if we switch to using parentheses, but let me know if this is premature and I'll figure out something simpler :).